at_functional_test/test/atclient_sync_conflict_test.dart

```diff
@@ -33,6 +32,31 @@ void main() async {
     }
     await Future.delayed(Duration(seconds: 10));
   });
+
+  // server value is null, client has non value for same key.
+  // you should get a sync conflict populated with no exception thrown.
+  test('server value is null', () async {
```

Comment:
The test is good. However, it did take me quite a few minutes of reading and re-reading the code before I could figure out what was going on. Please can you add some comments to explain what is going on?

Reply:
Updated the comments and pushed the changes.